plugins/kubernetesapi-report/src/main/java/com/freenow/sauron/plugins/commands/KubernetesGetDeploymentSpecCommand.java

```diff
@@ -0,0 +1,54 @@
+package com.freenow.sauron.plugins.commands;
+
+import com.freenow.sauron.plugins.utils.KubernetesResources;
+import io.kubernetes.client.openapi.ApiClient;
+import io.kubernetes.client.openapi.ApiException;
+import io.kubernetes.client.openapi.apis.*;
+import io.kubernetes.client.openapi.models.*;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.Optional;
+import java.util.Random;
+
+import static com.freenow.sauron.plugins.utils.KubernetesConstants.K8S_API_TIMEOUT_SECONDS;
+import static com.freenow.sauron.plugins.utils.KubernetesConstants.K8S_DEFAULT_NAMESPACE;
+import static com.freenow.sauron.plugins.utils.KubernetesConstants.K8S_PRETTY_OUTPUT;
+
+@Slf4j
+public class KubernetesGetDeploymentSpecCommand
+{
+    private static final Random RANDOM = new Random();
+
+    public Optional<V1DeploymentSpec> getDeployment(String serviceLabel, KubernetesResources resource, String service, ApiClient client)
+    {
+        try
+        {
+            String labelSelector = String.format("%s=%s", serviceLabel, service);
+            log.debug("Filtering deployment {} using selector {}", resource, labelSelector);
+            return new AppsV1Api(client).listNamespacedDeployment(
+                    K8S_DEFAULT_NAMESPACE,
+                    K8S_PRETTY_OUTPUT,
+                    false,
+                    null,
+                    null,
+                    labelSelector,
+                    null,
+                    null,
+                    null,
+                    K8S_API_TIMEOUT_SECONDS,
+                    false
+            ).getItems().stream().sorted(KubernetesGetDeploymentSpecCommand::random).findAny().map(V1Deployment::getSpec);
```

Comment:
As we discussed, here we need to get latest deployment, just to be safe in case there are two deployments available. Maybe something we can do by modifying the `sorted`.